### PR TITLE
Don't close long-lived RowSets in Downsampler BucketState

### LINKED ISSUE #5476: Downsample use-after-close issue
Reported from community slack:
```
aph-updateExecutor-3 | rumentedTableListenerBase | Uncaught exception for entry= downsample listener, added.size()=6, modified.size()=20533, removed.size()=0, shifted.size()=0, modifiedColumnSet={tdate_1min,tdate3_1min,tdate3_diff_1min,tdate2_1min,bar_volume_1min,total_volume_1min,bar_open_1min,bar_high_1min,bar_low_1min,bar_close_1min,opening_price_1min,Count_1min,yyyymmdd_1min,MonthOfYear_1min,DayOfYear_1min,WeekOfYear_1min,Open_Interest_1min,V_OI_1min,V_OI_Total_1min,prev_bar_open_1min,prev2_bar_open_1min,prev3_bar_open_1min,prev_bar_high_1min,prev2_bar_high_1min,prev3_bar_high_1min,prev_bar_low_1min,prev2_bar_low_1min,prev3_bar_low_1min,prev_bar_close_1min,prev2_bar_close_1min,prev3_bar_close_1min,prev_bar_volume_1min,prev2_bar_volume_1min,prev3_bar_volume_1min,prev_DayOfYear_1min,prev2_DayOfYear_1min,prev_WeekOfYear_1min,prev2_WeekOfYear_1min,prev_MonthOfYear_1min,prev2_MonthOfYear_1min,TypicalPrice_1min,prev_TypicalPrice_1min,prev2_TypicalPrice_1min,new_day_1min,new_week_1min,new_month_1min,di_plus_14_1min,di_minus_14_1min,adx_14_1min,stochRSI_fastk_14_1min,stochRSI_fastd_14_1min,prev_adx_14_1min,prev_di_plus_14_1min,prev_di_minus_14_1min,prev_stochRSI_fastk_14_1min,prev_stochRSI_fastd_14_1min,prev2_adx_14_1min,prev2_di_plus_14_1min,prev2_di_minus_14_1min,prev2_stochRSI_fastk_14_1min,prev2_stochRSI_fastd_14_1min,tdate_3min,tdate3_3min,tdate3_diff_3min,tdate2_3min,bar_volume_3min,total_volume_3min,bar_open_3min,bar_high_3min,bar_low_3min,bar_close_3min,opening_price_3min,Count_3min,yyyymmdd_3min,MonthOfYear_3min,DayOfYear_3min,WeekOfYear_3min,Open_Interest_3min,V_OI_3min,V_OI_Total_3min,prev_bar_open_3min,prev2_bar_open_3min,prev3_bar_open_3min,prev_bar_high_3min,prev2_bar_high_3min,prev3_bar_high_3min,prev_bar_low_3min,prev2_bar_low_3min,prev3_bar_low_3min,prev_bar_close_3min,prev2_bar_close_3min,prev3_bar_close_3min,prev_bar_volume_3min,prev2_bar_volume_3min,prev3_bar_volume_3min,prev_DayOfYear_3min,prev2_DayOfYear_3min,prev_WeekOfYear_3min,prev2_WeekOfYear_3min,prev_MonthOfYear_3min,...}:
java.lang.NullPointerException: Cannot invoke "io.deephaven.engine.rowset.impl.OrderedLongSet.ixGetRowSequenceIterator()" because "this.innerSet" is null
	at io.deephaven.engine.rowset.impl.WritableRowSetImpl.getRowSequenceIterator(WritableRowSetImpl.java:240)
	at io.deephaven.engine.rowset.RowSetShiftData.apply(RowSetShiftData.java:297)
	at io.deephaven.clientsupport.plotdownsampling.BucketState.shift(BucketState.java:113)
	at io.deephaven.clientsupport.plotdownsampling.RunChartDownsample$DownsamplerListener.handleShifts(RunChartDownsample.java:675)
	at io.deephaven.clientsupport.plotdownsampling.RunChartDownsample$DownsamplerListener.onUpdate(RunChartDownsample.java:332)
	at io.deephaven.engine.table.impl.InstrumentedTableUpdateListener$Notification.lambda$run$0(InstrumentedTableUpdateListener.java:37)
	at io.deephaven.engine.table.impl.InstrumentedTableListenerBase$NotificationBase.doRunInternal(InstrumentedTableListenerBase.java:334)
	at io.deephaven.engine.table.impl.InstrumentedTableListenerBase$NotificationBase.doRun(InstrumentedTableListenerBase.java:316)
	at io.deephaven.engine.table.impl.InstrumentedTableUpdateListener$Notification.run(InstrumentedTableUpdateListener.java:37)
	at io.deephaven.engine.updategraph.impl.BaseUpdateGraph.runNotification(BaseUpdateGraph.java:712)
	at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph$ConcurrentNotificationProcessor.processSatisfiedNotifications(PeriodicUpdateGraph.java:836)
	at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph$NotificationProcessorThreadFactory.lambda$newThread$0(PeriodicUpdateGraph.java:1109)
	at java.base/java.lang.Thread.run(Thread.java:1583)
aph-updateExecutor-3 | .AsyncClientErrorNotifier | Error in table update: java.lang.NullPointerException: Cannot invoke "io.deephaven.engine.rowset.impl.OrderedLongSet.ixGetRowSequenceIterator()" because "this.innerSet" is null
	at io.deephaven.engine.rowset.impl.WritableRowSetImpl.getRowSequenceIterator(WritableRowSetImpl.java:240)
	at io.deephaven.engine.rowset.RowSetShiftData.apply(RowSetShiftData.java:297)
	at io.deephaven.clientsupport.plotdownsampling.BucketState.shift(BucketState.java:113)
	at io.deephaven.clientsupport.plotdownsampling.RunChartDownsample$DownsamplerListener.handleShifts(RunChartDownsample.java:675)
	at io.deephaven.clientsupport.plotdownsampling.RunChartDownsample$DownsamplerListener.onUpdate(RunChartDownsample.java:332)
	at io.deephaven.engine.table.impl.InstrumentedTableUpdateListener$Notification.lambda$run$0(InstrumentedTableUpdateListener.java:37)
	at io.deephaven.engine.table.impl.InstrumentedTableListenerBase$NotificationBase.doRunInternal(InstrumentedTableListenerBase.java:334)
	at io.deephaven.engine.table.impl.InstrumentedTableListenerBase$NotificationBase.doRun(InstrumentedTableListenerBase.java:316)
	at io.deephaven.engine.table.impl.InstrumentedTableUpdateListener$Notification.run(InstrumentedTableUpdateListener.java:37)
	at io.deephaven.engine.updategraph.impl.BaseUpdateGraph.runNotification(BaseUpdateGraph.java:712)
	at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph$ConcurrentNotificationProcessor.processSatisfiedNotifications(PeriodicUpdateGraph.java:836)
	at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph$NotificationProcessorThreadFactory.lambda$newThread$0(PeriodicUpdateGraph.java:1109)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

It looks like the update graph is running and trying to apply a shift after the user released the downsampled table.

An older stack trace that looks like the same issue, achieved with the `RunChartDownsample.validate` flag being turned on:
```
ph-updateExecutor-29 | rumentedTableListenerBase | Uncaught exception for entry= downsample listener, added.size()=28, modified.size()=0, removed.size()=0, shifted.size()=0, modifiedColumnSet={}:
java.lang.NullPointerException: Cannot invoke "io.deephaven.engine.rowset.impl.OrderedLongSet.ixRangeIterator()" because "this.innerSet" is null
    at io.deephaven.engine.rowset.impl.WritableRowSetImpl.rangeIterator(WritableRowSetImpl.java:468)
    at io.deephaven.engine.rowset.impl.RowSetUtils.toString(RowSetUtils.java:24)
    at io.deephaven.engine.rowset.impl.WritableRowSetImpl.toString(WritableRowSetImpl.java:498)
    at java.base/java.lang.String.valueOf(String.java:4216)
    at java.base/java.io.PrintStream.println(PrintStream.java:1047)
    at io.deephaven.clientsupport.plotdownsampling.BucketState.validate(BucketState.java:313)
    at io.deephaven.clientsupport.plotdownsampling.RunChartDownsample$DownsamplerListener.lambda$onUpdate$1(RunChartDownsample.java:352)
    at java.base/java.lang.Iterable.forEach(Iterable.java:75)
    at io.deephaven.clientsupport.plotdownsampling.RunChartDownsample$DownsamplerListener.onUpdate(RunChartDownsample.java:352)
    at io.deephaven.engine.table.impl.InstrumentedTableUpdateListener$Notification.lambda$run$0(InstrumentedTableUpdateListener.java:37)
    at io.deephaven.engine.table.impl.InstrumentedTableListenerBase$NotificationBase.doRunInternal(InstrumentedTableListenerBase.java:334)
    at io.deephaven.engine.table.impl.InstrumentedTableListenerBase$NotificationBase.doRun(InstrumentedTableListenerBase.java:316)
    at io.deephaven.engine.table.impl.InstrumentedTableUpdateListener$Notification.run(InstrumentedTableUpdateListener.java:37)
    at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph.runNotification(PeriodicUpdateGraph.java:1322)
    at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph$ConcurrentNotificationProcessor.processSatisfiedNotifications(PeriodicUpdateGraph.java:1373)
    at io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph$NotificationProcessorThreadFactory.lambda$newThread$0(PeriodicUpdateGraph.java:1911)
    at org.jpy.PyLib.callAndReturnObject(Native Method)
    at org.jpy.PyObject.call(PyObject.java:449)
    at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
    at java.base/java.lang.Thread.run(Thread.java:833)
```

### PR DESCRIPTION
**Pull request by niloc132**
Fixes a bug in downsampling where a table may cause an error if freed while processing an update.

Fixes #5476

### CODE REVIEW

File: ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java [hunk 1]
```diff
@@ -310,22 +314,15 @@ public void validate(final boolean usePrev, final DownsampleChunkContext context
                         values[columnIndex].validate(offset, keyChunk.get(indexInChunk), valueChunks[columnIndex],
                                 indexInChunk, trackNulls ? nulls[columnIndex] : null);
                     } catch (final RuntimeException e) {
-                        System.out.println(rowSet);
                         final String msg =
                                 "Bad data! indexInChunk=" + indexInChunk + ", col=" + columnIndex + ", usePrev="
-                                        + usePrev + ", offset=" + offset + ", rowSet=" + keyChunk.get(indexInChunk);
+                                        + usePrev + ", offset=" + offset + ", indexInChunk=" + keyChunk.get(indexInChunk);
+                        log.error().append(msg).append("rowSet=").append(rowSet).endl();
```
Comment (rcaudy) [inline ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java:320]:
```suggestion
                        log.error().append(msg).append(", rowSet=").append(rowSet).endl();
```

File: ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java [hunk 2]
```diff
@@ -310,22 +314,15 @@ public void validate(final boolean usePrev, final DownsampleChunkContext context
...
                         throw new IllegalStateException(msg, e);
                     }
                 }
             }
         }
         Assert.eqTrue(makeRowSet().subsetOf(rowSet), "makeRowSet().subsetOf(rowSet)");
     }
-
-    public void close() {
```
Comment (rcaudy) [inline ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java:325]: Are you sure we shouldn't keep this part? I suppose it's safer...

**Review by rcaudy — COMMENTED** [review]
.

File: ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java (hunk 2 above)
Comment (niloc132) [reply ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java:325]: What would we want to use it for?

Comment (rcaudy) [reply ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java:325]: When we get rid of bucket states that we're recomputing?

**Review by rcaudy — APPROVED** [review]

File: ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java (hunk 2 above)
Comment (niloc132) [reply ClientSupport/src/main/java/io/deephaven/clientsupport/plotdownsampling/BucketState.java:325]: What I'm asking is "what would `BucketState.close()` do, if there is no expectation that long-lived RowSets should be closed"?

**Pull request merged**